src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -0,0 +1,427 @@
+
+    private static final Log LOGGER = LogFactory.getLog(ModsImporter.class);
+    private static final String KEYWORD_SEPARATOR = "; ";
+
+    private static final Pattern MODS_PATTERN = Pattern.compile("<mods .*>");
+
+
+    @Override
+    protected boolean isRecognizedFormat(BufferedReader input) throws IOException {
+        return input.lines().anyMatch(line -> MODS_PATTERN.matcher(line).find());
+    }
+
+    @Override
+    public ParserResult importDatabase(BufferedReader input) throws IOException {
+        Objects.requireNonNull(input);
+
+        List<BibEntry> bibItems = new ArrayList<>();
+
+        JAXBContext context;
+        try {
+            context = JAXBContext.newInstance("net.sf.jabref.logic.importer.fileformat.mods");
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+
+            //The unmarshalled object is a jaxbElement.
+            JAXBElement<?> unmarshalledObject = (JAXBElement<?>) unmarshaller.unmarshal(input);
+
+            //To get the root element we have to invoke getValue on the unmarshalled object
+            if (unmarshalledObject.getValue() instanceof ModsCollectionDefinition) {
+                ModsCollectionDefinition collection = (ModsCollectionDefinition) unmarshalledObject.getValue();
+                List<ModsDefinition> mods = collection.getMods();
+                parseModsCollection(bibItems, mods);
+            } else if (unmarshalledObject.getValue() instanceof ModsDefinition) {
+                ModsDefinition modsDefinition = (ModsDefinition) unmarshalledObject.getValue();
+                parseMods(bibItems, modsDefinition);
+            }
+        } catch (JAXBException e) {
+            LOGGER.debug("could not parse document", e);
+            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
+        }
+        return new ParserResult(bibItems);
+    }
+
+    private void parseModsCollection(List<BibEntry> bibItems, List<ModsDefinition> mods) {
+        for (ModsDefinition modsDefinition : mods) {
+            parseMods(bibItems, modsDefinition);
+        }
+    }
+
+    private void parseMods(List<BibEntry> bibItems, ModsDefinition modsDefinition) {
+        BibEntry entry = new BibEntry();
+        Map<String, String> fields = new HashMap<>();
+        if (modsDefinition.getID() != null) {
+            entry.setCiteKey(modsDefinition.getID());
+        }
+        if (modsDefinition.getModsGroup() != null) {
+            parseModsGroup(fields, modsDefinition.getModsGroup(), entry);
+        }
+        entry.setField(fields);
+        bibItems.add(entry);
+    }
+
+    private void parseModsGroup(Map<String, String> fields, List<Object> modsGroup, BibEntry entry) {
+        List<String> keywords = new ArrayList<>();
+        List<String> authors = new ArrayList<>();
+        List<String> notes = new ArrayList<>();
+
+        for (Object groupElement : modsGroup) {
+            if (groupElement instanceof AbstractDefinition) {
```

Comment:
Yeah, I know. The thing is, my idea/thought was to create a single method which does the instanceof checking in a generic way, like  `static <T> getElement(Obj groupElement, Class<T> clazz) `
Have a look at the second anwer here:
http://stackoverflow.com/questions/14228637/why-instanceof-does-not-work-with-generic?noredirect=1&lq=1

Then you could simply do a call: like  `GenreDefinition def = getElement(modsElement, GenreDefintion.class)`
And of course you would need to check for isAssignable